plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/StandardLanguage.xtend

```diff
@@ -0,0 +1,152 @@
+import org.eclipse.xtext.xtext.generator.xbase.XtypeGeneratorFragment2
+import org.eclipse.xtext.util.internal.Log
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ */
+@Accessors(PUBLIC_SETTER)
+@Log class StandardLanguage extends LanguageConfig2 {
+	
+	GrammarAccessFragment2 grammarAccess = new GrammarAccessFragment2
+	 
+	SerializerFragment2 serializer = new SerializerFragment2
+	
+	ResourceFactoryFragment2 resourceFactoryFragment = new ResourceFactoryFragment2
+
+	EMFGeneratorFragment2 emfGenerator = new EMFGeneratorFragment2
+	
+	XtextAntlrGeneratorFragment2 parserGenerator = new XtextAntlrGeneratorFragment2
+	
+	ValidatorFragment2 validator = new ValidatorFragment2 => [
+		addComposedCheck("org.eclipse.xtext.validation.NamesAreUniqueValidator")
+	]
+	
+	Formatter2Fragment2 formatter = new Formatter2Fragment2 
+	
+	GeneratorFragment2 generator = new GeneratorFragment2 
+	
+	BuilderIntegrationFragment2 builder = new BuilderIntegrationFragment2 
+	
+	ImportNamespacesScopingFragment2 scopeProvider = new ImportNamespacesScopingFragment2
+	
+	QualifiedNamesFragment2 qualifiedNamesProvider = new QualifiedNamesFragment2
+	
+	TypesGeneratorFragment2 commonTypesSupport = new TypesGeneratorFragment2 => [
+		onlyEnabledIfGrammarIsUsed = true
+	]
+	
+	XtypeGeneratorFragment2 xtypeSupport = new XtypeGeneratorFragment2
+	
+	XbaseGeneratorFragment2 xbaseSupport = new XbaseGeneratorFragment2
+	
+	Junit4Fragment2 junitSupport = new Junit4Fragment2
+	
+	QuickfixProviderFragment2 quickFixProvider =  new QuickfixProviderFragment2 
+	
+	LabelProviderFragment2 labelProvider = new LabelProviderFragment2 
+	
+	OutlineTreeProviderFragment2 outline = new OutlineTreeProviderFragment2 
+	
+	QuickOutlineFragment2 quickOutline = new QuickOutlineFragment2 
+	
+	CompareFragment2 compareEditor = new CompareFragment2 
+	
+	ContentAssistFragment2 contentAssist = new ContentAssistFragment2 
+	
+	RefactorElementNameFragment2 renameRefactoring = new RefactorElementNameFragment2
+	
+	CodetemplatesGeneratorFragment2 codeTemplates = new CodetemplatesGeneratorFragment2
+	 
+	XtextAntlrIDEAGeneratorFragment ideaParser = new XtextAntlrIDEAGeneratorFragment
+	
+	IdeaPluginGenerator ideaPlugin = new IdeaPluginGenerator
+	
+	WebIntegrationFragment webSupport = new WebIntegrationFragment => [
+		framework = "Ace"
+		generateServlet = true
+		generateJettyLauncher = true
+		generateHtmlExample = true
+	]
+	
+	new() {
+		try {
+			class.classLoader.loadClass("org.eclipse.xtext.xbase.XbaseRuntimeModule")
+			standaloneSetup = new XtextLanguageStandaloneSetup => [
+				addLoadedResource("platform:/resource/org.eclipse.xtext.xbase/model/Xbase.genmodel")
+			]
+		} catch (ClassNotFoundException e) {
+			LOG.info("Skipping registration of Xbase genmodel. Xbase is not on the classpath.")
+		}
+	}
+	
+	override initialize(Injector injector) {
+		val configuredFragments = new ArrayList(fragments)
+		fragments.clear
```

Comment:
and adds the later again. 
But I agree it's not nice. I could alternatively use fragments.add(i++, bla) starting with null, which I also didn't quite like.